Ray.jl/src/runtime.jl

```diff
@@ -181,16 +175,21 @@ function task_executor(ray_function, returns_ptr, task_args_ptr, task_name,
                                 rayjll.unwrap_function_descriptor(fd),
                                 get_current_job_id())
         # TODO: write generic Ray.put and Ray.get functions and abstract over this buffer stuff
-        args = map(task_args) do arg
+        flattened = map(task_args) do arg
             v = take!(rayjll.GetData(arg[]))
             io = IOBuffer(v)
             return deserialize(io)
         end
+        args, kwargs = recover_args(flattened)
 
-        arg_string = join(string.("::", typeof.(args)), ", ")
-        @info "Calling $func($arg_string)"
+        @info begin
+            args_str = join([string("::", typeof(arg)) for arg in args], ", ")
+            kwargs_str = join([string(k, "::", typeof(v)) for (k, v) in kwargs], ", ")
+            args_kwargs_str = join([args_str, kwargs_str], "; ")
+            "Calling $func($args_kwargs_str)"
```

Comment:
this seems more natural to me:
```suggestion
            "Calling $func($args_str; $kwargs_str)"
```

Reply:
I agree with you. I meant to get back to this as what I was trying to do was avoid having the "; " present when no keywords were used. I'll update the logic here to be:
```julia
isempty(kwargs_str) || (args_str *= "; " * kwargs_str)
"Calling $func($args_str)"
```

Reply:
could make `kwargs_str = isempty(kwargs) ? "" : string("; ", ...)`

Reply:
Unfortunately with that we end up with `$func($args_str$kwargs_str)`

Reply:
Ended up with this:
```julia
param_str = join((string("::", typeof(arg)) for arg in args), ", ")
if !isempty(kwargs)
    param_str *= "; "
    param_str *= join((string(k, "::", typeof(v)) for (k, v) in kwargs), ", ")
end
"Calling $func($param_str)"
```
Having `param_str *= "; "` on a separate line just avoids hitting the line length limit